content/200-concepts/100-components/02-prisma-client/058-transactions.mdx

```diff
@@ -87,6 +87,85 @@ Instead of immediately awaiting the result of each operation when it's performed
 >
 > Refer to the 📖 [transactions guide](../../../guides/performance-and-optimization/prisma-client-transactions-guide#transaction-api) for more examples.
 
+### Interactive Transactions
+
+Sometimes you need more control over what queries execute within a transaction.
+
+Interactive transactions are commonly used when the result of one query depends on another query and you want to run both queries within a single transaction.
+
+To support this use case and more, you can pass an async function into `$transaction`.
+
+The following is an example of purchasing a movie ticket online. This is a good use case for an interactive transactions because you don't want to sell the same ticket twice.
+
+```tsx
+import { Ticket } from 'prisma'
+
+const userEmail = 'alice@prisma.io'
+const movieName = 'October Sky'
+
+// Start a transaction
+const ticket: Ticket = await prisma.$transaction(async (prisma) => {
+  // Find an available ticket
+  const availableTicket = await prisma.ticket.findFirst({
+    where: {
+      claimedBy: null,
+      movie: {
+        name: movieName,
+      },
+    },
+  })
+
+  // Throw an error if no tickets are available
+  if (!availableTicket) {
+    // This will rollback the transaction
+    throw new Error(`Oh no! ${movieName} is all booked.`)
+  }
+
+  // Claim the ticket, commit the transaction, and return
+  // the result
+  return prisma.ticket.update({
+    data: {
+      claimedBy: userEmail,
+    },
+    where: {
+      id: availableTicket.id,
+    },
+  })
+})
+```
+
+In the example above, both the `findFirst` and `update` queries run within a database transaction. When your application reaches the end of the function, the transaction is **committed** to the database.
+
+If your application encounters an error along the way, the async function will throw an exception and automatically **rollback** the transaction. To catch the exception, you can wrap `$transaction` in a try-catch block.
```

Comment:
I think it would be helpful to show the try catch block. being as clear as possible will reduce the risk of people misunderstanding from a copy paste